pyuvsim/telescope.py

```diff
@@ -37,3 +41,278 @@ def __eq__(self, other):
+                     'reff': 'ref_freq', 'ind': 'spectral_index'}
+
+    string_mode = True
+
+    def __init__(self, beam_list=None, uvb_params=None):
+
+        self.uvb_params = {'freq_interp_kind': 'cubic',
+                           'interpolation_function': 'az_za_simple'}
+
+        self._str_beam_list = []
+        self._obj_beam_list = []
+        if beam_list is not None:
+            if all([isinstance(b, str) for b in beam_list]):
+                self._str_beam_list[:] = beam_list[:]
+                self.string_mode = True
+            elif all([not isinstance(b, str) for b in beam_list]):
+                self._obj_beam_list[:] = beam_list[:]
+                self.string_mode = False
+            else:
+                raise ValueError("Invalid beam list: " + str(beam_list))
+
+        # If any UVBeam objects are passed in, update uvb_params:
+        list_uvb_params = {}
+        if uvb_params is None:
+            uvb_params = {}
+            if not self.string_mode:
+                list_uvb_params = self._scrape_uvb_params(self._obj_beam_list, strict=False)
+        self.uvb_params.update(list_uvb_params)
+        self.uvb_params.update(uvb_params)    # Optional parameters for the UVBeam objects
+
+    def _scrape_uvb_params(self, beam_objs, strict=True):
+        """
+        Collect uvb_params from the set of beam objects.
+
+        Parameters
+        ----------
+        beam_objs : list
+            If any UVBeams are found in this list, they will be scraped
+            for relevant parameters.
+
+        strict : bool
+            If True, will raise an error if the UVBeams in beam_objs
+            have conflicting parameters.
+
+        Notes
+        -----
+        If beam_objs has strings in it, this will simply return an empty dict.
+        """
+
+        new_pars = {}
+        for bm in beam_objs:
+            if isinstance(bm, UVBeam):
+                for key in self.uvb_params.keys():
+                    val = getattr(bm, key)
+                    if key not in new_pars:
+                        new_pars[key] = []
+                    if val is not None:
+                        new_pars[key].append(val)
+        for key, vals in new_pars.items():
+            svals = set(vals)
+            if len(svals) == 1:
+                new_pars[key] = svals.pop()
+            elif len(svals) > 1:
+                try:
+                    raise ValueError('Conflicting settings for {}: {}'.format(key, str(svals)))
+                except ValueError as err:
+                    if strict:
+                        raise err
+                    else:
+                        warnings.warn(err.args[0])  # Use exception message as warning message.
+
+        new_pars = {k: v for k, v in new_pars.items() if not v == []}
+
+        return new_pars
+
+    def _set_params_on_uvbeams(self, beam_objs):
+        """
+        Set parameters on UVBeam objects using uvb_params.
```

Comment:
This docstring will error with flake8 when we turn on docstring testing in Steven's PR -- One line docstrings should fit on one line including quotes.